src/alire/alire-crate_configuration.adb

```diff
@@ -0,0 +1,401 @@
+with Ada.Strings.Unbounded; use Ada.Strings.Unbounded;
+with Ada.Directories;
+with Ada.Text_IO;
+with Ada.Characters.Handling;
+
+with Alire_Early_Elaboration;
+with Alire.Solutions;
+with Alire.Roots;
+with Alire.Utils.TTY;
+with Alire.Origins;
+
+with GNAT.IO;
+
+with Alire.Directories;
+
+with TOML; use TOML;
+
+package body Alire.Crate_Configuration is
+
+   package TTY renames Utils.TTY;
+   package TIO renames Ada.Text_IO;
+
+   ----------
+   -- Load --
+   ----------
+
+   procedure Load (This : in out Global_Config;
+                   Root :        Alire.Roots.Root)
+   is
+      Solution : constant Solutions.Solution := Root.Solution;
+   begin
+
+      --  Warnings when setting up an incomplete environment
+
+      if not Solution.Is_Complete then
+         Trace.Debug ("Generating incomplete environment"
+                      & " because of missing dependencies");
+
+         --  Normally we would generate a warning, but since that will pollute
+         --  the output making it unusable, for once we write directly to
+         --  stderr (unless quiet is in effect):
+
+         if not Alire_Early_Elaboration.Switch_Q then
+            GNAT.IO.Put_Line
+              (GNAT.IO.Standard_Error,
+               TTY.Warn ("warn:") & " Generating incomplete environment"
+               & " because of missing dependencies");
+         end if;
```

Comment:
Does this output to stderr make sense here and in `80:92`, or would a `Trace.Warning` be enough? I don't remember any parseable output being generated for this feature?

Reply:
I think this is copy/paste from legacy code.

Reply:
Actually there is a comment above:
```
         --  Normally we would generate a warning, but since that will pollute
         --  the output making it unusable, for once we write directly to
         --  stderr (unless quiet is in effect):
```
I think this is coming from you. I don't know if it is still applicable.

Reply:
Yes, I remember writing that. That's because when generating the output of `printenv`, our logs should not intermingle with the intended output. If this has no relation, it can be replaced with a mere `Trace.Warning`.

Reply:
Fixed in #699.